src/Symfony/Component/Serializer/Normalizer/GetSetMethodNormalizer.php

```diff
@@ -47,7 +47,7 @@ public function denormalize($data, $class, $format = null, array $context = arra
         $normalizedData = $this->prepareForDenormalization($data);
 
         $reflectionClass = new \ReflectionClass($class);
-        $object = $this->instantiateObject($normalizedData, $class, $context, $reflectionClass, $allowedAttributes);
+        $object = $this->instantiateComplexObject($normalizedData, $class, $context, $reflectionClass, $allowedAttributes, $format);
```

Comment:
An alternative approach to let the user change it easily, would be to create a protected function `getInstantiateObjectMethod(): string`, which defaults to `instantiateObject()` but can easily be overridden to return `instantiateComplexObject()` instead. This would allow the users to easily switch to the new method without too much work.